From 0ae410d94910213567da10caa498ff1b1aa32887 Mon Sep 17 00:00:00 2001
From: Xiangyou Xie <xiexiangyou@huawei.com>
Date: Fri, 19 Feb 2021 15:14:31 +0800
Subject: [PATCH] cpuidle-haltpoll: Use arch_cpu_idle() to replace
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 0ae410d94910213567da10caa498ff1b1aa32887

 default_idle()

hulk inclusion
category: feature
bugzilla: 47727
CVE: NA

------------------------------

Use arch_cpu_idle() to replace default_idle() in default_enter_idle().
default_idle() is defined only in x86.

Signed-off-by: Xiangyou Xie <xiexiangyou@huawei.com>
Reviewed-by: Hanjun Guo <guohanjun@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/x86/kernel/process.c          | 1 +
 drivers/cpuidle/cpuidle-haltpoll.c | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/x86/kernel/process.c b/arch/x86/kernel/process.c
index 145a7ac0c19a..d7f23d1c7391 100644
--- a/arch/x86/kernel/process.c
+++ b/arch/x86/kernel/process.c
@@ -689,6 +689,7 @@ void __cpuidle default_idle(void)
 }
 #if defined(CONFIG_APM_MODULE) || defined(CONFIG_HALTPOLL_CPUIDLE_MODULE)
 EXPORT_SYMBOL(default_idle);
+EXPORT_SYMBOL(arch_cpu_idle);
 #endif
 
 #ifdef CONFIG_XEN
diff --git a/drivers/cpuidle/cpuidle-haltpoll.c b/drivers/cpuidle/cpuidle-haltpoll.c
index fcc53215bac8..f2fc7204de54 100644
--- a/drivers/cpuidle/cpuidle-haltpoll.c
+++ b/drivers/cpuidle/cpuidle-haltpoll.c
@@ -32,7 +32,7 @@ static int default_enter_idle(struct cpuidle_device *dev,
 		local_irq_enable();
 		return index;
 	}
-	default_idle();
+	arch_cpu_idle();
 	return index;
 }
 
-- 
2.26.2

